zproject/backends.py

```diff
@@ -2845,7 +2845,7 @@ def slo_request_to_idp(
 
         idp_name = cls.get_logged_in_user_idp(request)
         if idp_name is None:
-            return None
+            raise AssertionError("User not logged in via SAML")
```

Comment:
Unless I'm missing something, can't this cleanup be folded back into the initial commit?  It's not related to per-IDP configuration of IDP SLO?

Reply:
This commit changes the intent of this function a bit, and this is a part of it. In the initial commit, this function gets called on the request, and is meant to check if the user logged in via SAML - if not, return `None` and the caller proceeds with normal logout.

In the commit changing this to `AssertionError`, this function changes intent and is meant to only be called after it has been verified that the user logged in via SAML (by calling `should_do_saml_sp_initiated_logout(request)` first). There's a good argument to be made that all this stuff should just be squashed into one commit with the proper, complete implementation - but it'll be kind of huge, and at least for the review stage I decided to keep these sequential commits, especially since folks have already looked through the earlier implementation steps so it may be useful to just have new commits on top, to easily be able to see what has changed.